### PR TITLE
ADD csubs control scripts

### PR DESCRIPTION
**Pull request by fgalan**
Tested with a copy of the orion.lab DB except some functionality (i.e. the one related with CB restart) that will be deferred to real testing on the orion.lab testbed.

### CODE REVIEW

File: scripts/managedb/csubs_control/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+This directory includes some scripts to help to control context subscriptions collection (csubs). All them share
```
Comment (kzangeli) [inline scripts/managedb/csubs_control/README.md]: All **of** them    OR

They all share ...

File: scripts/managedb/csubs_control/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+the same CLI arguments:
+
+* `-db <database>` (mandatory): to specify the DB to process
+* `--dry-run (optional): it avoids actual modification in the DB, just reporting
+* `--with-restart <status_file>` (optional): restart Context Broker service if the database was manipulated and,
+   consecuently, Context Broker needs to be restarted. It is based in `/etc/init.d/contextBroker` service script,
```
Comment (kzangeli) [inline scripts/managedb/csubs_control/README.md]: based on

File: scripts/managedb/csubs_control/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+   so it will not work if the CB was not installed as RPM. It requires root permissions.
+* `-v` (optional): enables verbose mode
+
+The scripts are:
+
+* csub_dups_checker.py: removes csubs duplicates. Have a look to script implementation in order to know which
```
Comment (kzangeli) [inline scripts/managedb/csubs_control/README.md]: Have a look at

File: scripts/managedb/csubs_control/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  csub fields are considered part of the 'signature' of the subscription that is used to detect duplicates.
+* csub_localhost_reference_checker.py: it checks subscriptions using "localhost" in the callback, removing them
+  if they are not allowed (based on the result of the valid_localhost_url() function; you would need to modify it
+  to implement your own checkings).
```
Comment (kzangeli) [inline scripts/managedb/csubs_control/README.md]: checkings => checks

File: scripts/managedb/csubs_control/README.md (hunk 3 above)
Comment (fgalan) [inline scripts/managedb/csubs_control/README.md]: Fixed in https://github.com/telefonicaid/fiware-orion/commit/7115621dcc90ba36dee2235ad4ccc909ee1b11d3

File: scripts/managedb/csubs_control/README.md (hunk 4 above)
Comment (fgalan) [inline scripts/managedb/csubs_control/README.md]: Fixed in https://github.com/telefonicaid/fiware-orion/commit/7115621dcc90ba36dee2235ad4ccc909ee1b11d3

File: scripts/managedb/csubs_control/README.md (hunk 2 above)
Comment (fgalan) [inline scripts/managedb/csubs_control/README.md]: Fixed in https://github.com/telefonicaid/fiware-orion/commit/7115621dcc90ba36dee2235ad4ccc909ee1b11d3

File: scripts/managedb/csubs_control/README.md (hunk 1 above)
Comment (fgalan) [inline scripts/managedb/csubs_control/README.md]: Fixed in https://github.com/telefonicaid/fiware-orion/commit/7115621dcc90ba36dee2235ad4ccc909ee1b11d3

**Comment by kzangeli** [comment]
LGTM

File: scripts/managedb/csubs_control/csub_dups_checker.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+# https://github.com/telefonicaid/fiware-orion/blob/develop/doc/manuals/admin/database_model.md#csubs-collection
+
+# Firs stage: analyze csubs documents
+duplicates = {}
+
+# Note that sorting by {expiration: -1} will get subscriptions by decreasing expiration order, thus
+# in the case of deleting a duplicated document, the first one in the duplicated set (i.e. the
+# one with higher expiration time) is the one that is preserved. Using a different ordering order based
+# on _id we could change the policy to preserve the first csub created
+n = 0
```
Comment (kzangeli) [inline scripts/managedb/csubs_control/csub_dups_checker.py]: Should we not want to remove all but the one with the newest  **Creation Time** ?

Comment (fgalan) [inline scripts/managedb/csubs_control/csub_dups_checker.py]: I have the doubt between keep the "last created one" (using creation time) or "the one that will last longer, no matter when was created" (using expiration). At the end, I chose the last criteria, which I think is better (if csubs are equal, it doesn't matter which one where created first).

Comment (kzangeli) [inline scripts/managedb/csubs_control/csub_dups_checker.py]: That might be true, but the problem we have is that the client "wants to do a sub-update but instead creates a new one". I really believe the last request (newest creation time) should be what is kept.

Not that it's really important ... I imagine (read: hope) this is just a fix until we implement this intelligence inside the broker

**Pull request merged**